Lib/sndhdr.py

```diff
@@ -158,13 +162,17 @@ def test_wav(h, f):
     # 'RIFF' <len> 'WAVE' 'fmt ' <len>
     if not h.startswith(b'RIFF') or h[8:12] != b'WAVE' or h[12:16] != b'fmt ':
         return None
-    f.seek(0)
     try:
+        f.seek(0)
         w = wave.open(f, 'r')
-    except (EOFError, wave.Error):
+    except (EOFError, RuntimeError, AttributeError, wave.Error):
```

Comment:
Catching AttributeError or RuntimeError *here* is wrong. wave.open() should be modified to better handle errors.

Reply:
If I remove the `RuntimeError` and `Attribute Error` from the catch list, this is what happens in the python run. 

```python
david@lithium ~/P/P/cpython>pso-34088>./python.exe                              10:40:56
Python 3.8.0a0 (heads/master:28f07364f0, Jul 17 2018, 11:42:07)
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sndhdr
>>> sndhdr.what('./Lib/test/sndhdrdata/input1.bad')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/david/Projects/PYTHON/cpython/Lib/sndhdr.py", line 54, in what
    res = whathdr(filename)
  File "/Users/david/Projects/PYTHON/cpython/Lib/sndhdr.py", line 63, in whathdr
    res = tf(h, f)
  File "/Users/david/Projects/PYTHON/cpython/Lib/sndhdr.py", line 162, in test_wav
    w = wave.open(f, 'r')
  File "/Users/david/Projects/PYTHON/cpython/Lib/wave.py", line 510, in open
    return Wave_read(f)
  File "/Users/david/Projects/PYTHON/cpython/Lib/wave.py", line 164, in __init__
    self.initfp(f)
  File "/Users/david/Projects/PYTHON/cpython/Lib/wave.py", line 153, in initfp
    chunk.skip()
  File "/Users/david/Projects/PYTHON/cpython/Lib/chunk.py", line 160, in skip
    self.file.seek(n, 1)
  File "/Users/david/Projects/PYTHON/cpython/Lib/chunk.py", line 113, in seek
    raise RuntimeError
RuntimeError
>>>
```


--- 


I don't have a complete understand what's happening inside `chunk.py`, it looks like seek is being asked to go too far in the file and it's throwing a runtime error.

I added a quick print to seek and this is what came out of my test. 
  
#### my change 
```python 
    def seek(self, pos, whence=0):
        """Seek to specified position into the chunk.
        Default position is 0 (start of chunk).
        If the file is not seekable, this will result in an error.
        """

        if self.closed:
            raise ValueError("I/O operation on closed file")
        if not self.seekable:
            raise OSError("cannot seek")
        if whence == 1:
            pos = pos + self.size_read
        elif whence == 2:
            pos = pos + self.chunksize
        if pos < 0 or pos > self.chunksize:
            print('SEEKING TO POSITION: ', pos, "CHUNK SIZE:", self.chunksize )
            raise RuntimeError
        self.file.seek(self.offset + pos, 0)
        self.size_read = pos
```


#### output of that test
```python
Python 3.8.0a0 (heads/master:28f07364f0, Jul 17 2018, 11:42:07)
[Clang 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sndhdr
>>> sndhdr.what('Lib/test/sndhdrdata/input1.bad')
SEEKING TO POSITION:  808464444 CHUNK SIZE: 808464432  ## pos > self.chunksize
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/david/Projects/PYTHON/cpython/Lib/sndhdr.py", line 54, in what
    res = whathdr(filename)
# . . .  
```



given that `skip()` passes on `OSError`, maybe it makes sense to ignore the `RuntimeError` from `seek()`?

Reply:
You are free to decide how to modify the code, but IMHO wave constructor should be modified to raise a wave.Error if the file is invalid.